Spec/RealtimeClientPresence.swift

```diff
@@ -316,6 +380,426 @@ class RealtimeClientPresence: QuickSpec {
                     expect(channel2.presenceMap.members).to(haveCount(2))
                 }
+
+                // RTP5c
+                context("when a channel becomes ATTACHED") {
+
+                    // RTP5c1
+                    it("if the resumed flag is true and no SYNC is initiated as part of the attach then do nothing, PresenceMap is not affected and no members need to be re-entered") {
+                        let options = AblyTests.commonAppSetup()
+
+                        var clientMembers: ARTRealtime?
+                        waitUntil(timeout: testTimeout) { done in
+                            clientMembers = AblyTests.addMembersSequentiallyToChannel("foo", members: 3, options: options) {
+                                done()
+                            }
+                        }
+                        defer { clientMembers?.dispose(); clientMembers?.close() }
+
+                        options.disconnectedRetryTimeout = 1.0
+                        options.tokenDetails = getTestTokenDetails(key: options.key!, ttl: 5.0)
+                        let client = AblyTests.newRealtime(options)
+                        defer { client.dispose(); client.close() }
+                        let channel = client.channels.get("foo")
+
+                        var originalMembers: [ARTPresenceMessage]?
+                        waitUntil(timeout: testTimeout) { done in
+                            channel.presence.get { members, error in
+                                expect(error).to(beNil())
+                                expect(members).to(haveCount(3))
+                                originalMembers = members
+                                done()
+                            }
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            let partialDone = AblyTests.splitDone(2, done: done)
+                            client.connection.once(.Disconnected) { stateChange in
+                                // Token expired
+                                partialDone()
+                            }
+                            client.connection.once(.Connecting) { stateChange in
+                                partialDone()
+                            }
+                        }
+
+                        guard let transport = client.transport as? TestProxyTransport else {
+                            fail("TestProxyTransport is not set"); return
+                        }
+
+                        waitUntil(timeout: testTimeout) { done in
+                            let partialDone = AblyTests.splitDone(2, done: done)
+                            transport.beforeProcessingReceivedMessage = { protocolMessage in
+                                if protocolMessage.action == .Attached {
+                                    // No Sync
+                                    expect(protocolMessage.hasPresence).to(beFalse())
```

Comment:
RTPC1 states:

>  If the resumed flag is true and no SYNC is initiated as part of the attach, do nothing as there is no loss of continuity on the channel and no change to presence.

You are testing it to be false.  You may need to inject an ATTACHED ProtocolMessage with has presence flag true for now as the realtime system does not adhere to this yet, hopefully by the end of this week it will be in sandbox so you can perhaps add a TODO to remove the fabricated message.